driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -1311,7 +1311,8 @@ protected void initChannel(SocketChannel channel) throws Exception {
             ChannelPipeline pipeline = channel.pipeline();
 
             if (sslOptions != null) {
-                pipeline.addLast("ssl", sslOptions.newSSLHandler(channel));
+                InetSocketAddress addr = new InetSocketAddress(connection.address.getAddress(), connection.address.getPort());
```

Comment:
Ah, I didn't realize InetSocketAddress' fields are 'immutable' so I took this approach to avoid possible side effects of state change. You are right though, it's unnecessary.